framework/validators/CompareValidator.php

```diff
@@ -85,28 +85,28 @@ public function init()
         if ($this->message === null) {
             switch ($this->operator) {
                 case '==':
-                    $this->message = Yii::t('yii', '{attribute} must be repeated exactly.');
+                    $this->message = Yii::t('yii', '{attribute} must be equal to "{compareInternal}".');
```

Comment:
> Why not keeping using {compareValue}?

@qiangxue, see this please: https://github.com/yiisoft/yii2/pull/7632#issuecomment-77773980.
Also, It would turn very confused the using of placeholders. There is already a `{compareValue}` placeholder and, as its name suggests, should be replace by a value, not a attribute name, and vice-versa (the issue here). This is why (I think) we need a third internal placeholder. All users' custom messages would continue to work ok. Only the default message would be adapted as using compareValue or compareAttribute params.

> Also, when you use compare validator to ensure password is repeated the same, this new message doesn't read very naturally and will almost always be changed.

Thanks, i will fix this idiomatic question. I did that way because, in portuguese, to say that A should be equal to B sounds much more natural (and clearer) than A should be repeated exactly (where?).